python/python/glide/async_commands/core.py

```diff
@@ -132,20 +132,124 @@ class InfBound(Enum):
     NEG_INF = "-inf"
 
 
-class ScoreLimit:
+class LexInfBound(Enum):
     """
-    Represents a score limit in a sorted set.
+    Enumeration representing lexicographic positive and negative infinity bounds for sorted set scores.
+    """
+
+    POS_INF = "+"
+    NEG_INF = "-"
+
+
+class ScoreBoundary:
+    """
+    Represents a score boundary in a sorted set.
 
     Args:
         value (float): The score value.
-        is_inclusive (bool): Whether the score value is inclusive. Defaults to False.
+        is_inclusive (bool): Whether the score value is inclusive. Defaults to True.
     """
 
     def __init__(self, value: float, is_inclusive: bool = True):
-        """Convert the score limit to the Redis protocol format."""
+        """Convert the score boundary to the Redis protocol format."""
         self.value = str(value) if is_inclusive else f"({value}"
 
 
+class LexBoundary:
+    """
+    Represents a lexicographic boundary in a sorted set.
+
+    Args:
+        value (str): The lex value.
+        is_inclusive (bool): Whether the score value is inclusive. Defaults to True.
+    """
+
+    def __init__(self, value: str, is_inclusive: bool = True):
+        """Convert the lexicographic boundary to the Redis protocol format."""
+        self.value = f"[{value}" if is_inclusive else f"({value}"
+
+
+class Limit:
+    """
+    Represents a limit argument for a range query in a sorted set to be used in ZRANGE command.
+
+    The optional LIMIT argument can be used to only get a range of the matching elements
+        (similar to SELECT LIMIT offset, count in SQL).
+    Args:
+        offset (int): The offset from the start of the range.
+        count (int): The number of elements to include in the range.
+            A negative count returns all elements from the offset.
+    """
+
+    def __init__(self, offset: int, count: int):
+        self.offset = offset
+        self.count = count
+
+
+class RangeByIndex:
+    """
+    Represents a range query by index (rank) in a sorted set.
+
+    The `start` and `stop` arguments represent zero-based indexes.
+    If `start` is greater than either the end index of the sorted set or `stop`, an empty list is returned.
```

Comment:
It's confusing that it says 'as empty list is returned' as a doc for the init function of this class, which obviously doesn't return a list:). 
Lets remove lines 194-195 as it depends on the server implementation, and instead add a link to the relevant documentation in redis